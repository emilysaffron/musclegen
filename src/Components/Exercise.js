import styled from "@emotion/styled";
import ToggleModal from "../Helpers/ToggleModal";
const StyledExercise = styled.button`
  background: #a8bbce;
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: 5px;
  margin: 5px;
  align-items: center;
  border-radius: 20px;
  width: 385px;
  height: 50px;
  &:hover {
    cursor: pointer;
  }
`;

const Exercise = ({
  name,
  target,
  toggleModal,
  modal,
  updateChosenExercise,
}) => {
  const HandleClick = () => {
    ToggleModal(toggleModal, modal);
    updateChosenExercise(name);
  };

  return (
    <StyledExercise onClick={HandleClick}>
      <div>{name}</div>
      <div>Target: {target}</div>
    </StyledExercise>
  );
};
export default Exercise;
