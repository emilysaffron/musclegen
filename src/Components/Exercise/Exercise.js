import styled from "@emotion/styled";
const StyledExercise = styled.button`
  background: #a8bbce;
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: 5px;
  margin: 5px;
  align-items: center;
  border-radius: 20px;
  width: 385px;
  height: 50px;
  cursor: pointer;
`;

const Exercise = ({
  name,
  target,
  toggleModal,
  modal,
  updateChosenExercise,
}) => {
  const HandleClick = () => {
    toggleModal(!modal);
    updateChosenExercise(name);
  };

  return (
    <StyledExercise onClick={HandleClick}>
      <div>{name}</div>
      <div>Target: {target}</div>
    </StyledExercise>
  );
};
export default Exercise;
