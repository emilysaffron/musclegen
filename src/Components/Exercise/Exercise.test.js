import renderer from "react-test-renderer";
import Exercise from "./Exercise";

it("Exercise Snapshot", () => {
  const tree = renderer
    .create(
      <Exercise
        name=""
        target=""
        toggleModal="toggleModal"
        modal={true}
        updateChosenExercise="updateChosenExercise"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
