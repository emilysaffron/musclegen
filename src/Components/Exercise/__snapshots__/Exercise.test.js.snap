// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exercise Snapshot 1`] = `
<button
  className="css-1tmfx1g"
  onClick={[Function]}
>
  <div>
    
  </div>
  <div>
    Target: 
    
  </div>
</button>
`;
